modules/account-lib/src/coin/sol/transaction.ts

```diff
@@ -241,16 +242,32 @@ export class Transaction extends BaseTransaction {
           });
           break;
         case InstructionBuilderTypes.Transfer:
-          inputs.push({
-            address: instruction.params.fromAddress,
-            value: instruction.params.amount,
-            coin: this._coinConfig.name,
-          });
-          outputs.push({
-            address: instruction.params.toAddress,
-            value: instruction.params.amount,
-            coin: this._coinConfig.name,
-          });
+          if (instruction.params.mint) {
+            const coin = getSolTokenFromAddress(instruction.params.mint, this._coinConfig.network);
+            assert(coin);
+            inputs.push({
+              address: instruction.params.fromAddress,
+              value: instruction.params.amount,
+              coin: coin.name,
+            });
+            outputs.push({
+              address: instruction.params.toAddress,
+              value: instruction.params.amount,
+              coin: coin.name,
+            });
+          } else {
+            inputs.push({
+              address: instruction.params.fromAddress,
+              value: instruction.params.amount,
+              coin: this._coinConfig.name,
+            });
+            outputs.push({
+              address: instruction.params.toAddress,
+              value: instruction.params.amount,
+              coin: this._coinConfig.name,
+            });
+          }
```

Comment:
duplicate code, refactor out coin name, eg.

```suggestion
            coinName = coin.name;
          } else {
            coinName = this._coinConfig.name;
          }
            inputs.push({
              address: instruction.params.fromAddress,
              value: instruction.params.amount,
              coin: coinName,
            });
            outputs.push({
              address: instruction.params.toAddress,
              value: instruction.params.amount,
              coin: coinName,
            });
          }
```